frappe/core/doctype/file/file_list.js

```diff
@@ -81,7 +81,35 @@ frappe.listview_settings['File'] = {
 		doclist.page.add_menu_item(__("Edit Folder"), function() {
 			frappe.set_route("Form", "File", doclist.current_folder);
 		});
-	},
+
+	doclist.page.add_menu_item(__("Import Zip-Archive"), function() {
+			// make upload dialog
+			frappe.ui.get_upload_dialog({
+				"args": {
+					"folder": doclist.current_folder,
+					"from_form": 1
```

Comment:
@rmehta 
ok, I can pass folder in frappe.ui.get_upload_dialog.

Which dynamic link you need in dialog?